src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+        return False
+
+    def __iter__(self):
+        """
+        Return an iter version of ``self``.
+        """
+        for e in self.groundset():
+            yield self(e)
+
+    def _repr_(self):
+        """
+        Return a string of the signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: from sage.matroids.oriented_matroids.signed_subset_element import SignedSubsetElement
+            sage: C = [((1,4),(2,3)), ((2,3),(1,4))]
+            sage: M = OrientedMatroid(C, key='circuit')
+            sage: SignedSubsetElement(M, data=((1,4),(2,3)))
+            +: 1,4
+            -: 2,3
+            0:
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: OrientedMatroid.options.display = 'vector'
+            sage: SignedSubsetElement(M, data=((1,4),(2,3)))
+            (1,-1,-1,1)
+            sage: OrientedMatroid.options.display = 'set'
+        """
+        from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        if OrientedMatroid.options.display == 'set':
+            p = map(str, self.positives())
+            n = map(str, self.negatives())
+            z = map(str, self.zeros())
+            return "+: " + ','.join(p) + "\n" + \
+                "-: " + ','.join(n) + "\n" +\
+                "0: " + ','.join(z)
+        if OrientedMatroid.options.display == 'vector':
+            return "(" + ','.join([str(self(e)) for e in self.groundset()]) + ")"
+
+    def _latex_(self):
+        r"""
+        Return a latex representation of the signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: from sage.matroids.oriented_matroids.signed_subset_element import SignedSubsetElement
+            sage: C = [((1,4),(2,3)), ((2,3),(1,4))]
+            sage: M = OrientedMatroid(C, key='circuit')
+            sage: latex(SignedSubsetElement(M, data=((1,4),(2,3))))
+            \left( \left{1,4\right},\left{2,3\right} \right)
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: OrientedMatroid.options.display = 'vector'
+            sage: latex(SignedSubsetElement(M, data =((1,4),(2,3))))
+            \left(1,-1,-1,1\right)
+        """
+        from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        if OrientedMatroid.options.display == 'set':
+            p = map(str, self.positives())
+            n = map(str, self.negatives())
+            return "\\left( \\left{" + ','.join(p) + \
+                "\\right},\\left{" + ','.join(n) + "\\right} \\right)"
+        if OrientedMatroid.options.display == 'vector':
+            ground_set = [str(self(e)) for e in self.groundset()]
+            return "\\left(" + ','.join(ground_set) + "\\right)"
+
+    def __copy__(self):
```

Comment:
These are (functionally) immutable, right? I would just have it return `self` because of this.

Reply:
I don't know what you mean by that? Isn't this supposed to return the string that it will output?

Reply:
I think he is referring to the `__copy__` and `__deepcopy__` methods; they should simply `return self` as in `matroid.pyx`.

Reply:
Indeed, that is what I am suggesting.